types/time_test.go

```diff
@@ -54,22 +97,27 @@ func (s *testTimeSuite) TestDateTime(c *C) {
 		{"121231113045", "2012-12-31 11:30:45"},
 		{"2012-02-29", "2012-02-29 00:00:00"},
 		{"00-00-00", "0000-00-00 00:00:00"},
-		{"00-00-00 00:00:00.123", "2000-00-00 00:00:00.123"},
+		{"00-00-00 00:00:00.123", "2000-00-00 00:00:00.123000"},
```

Comment:
I wonder why the FSP of `00-00-00 00:00:00.123` is 6 and `1701020304.1` is 0 ?
Would like to explain it ?

Reply:
mysql> select @@sql_mode;
+------------+
| @@sql_mode |
+------------+
|            |
+------------+
1 row in set (0.00 sec)

mysql> select cast("1701020302.11" as datetime);
+-----------------------------------+
| cast("1701020302.11" as datetime) |
+-----------------------------------+
| 2017-01-02 03:02:11               |
+-----------------------------------+
1 row in set (0.00 sec)
mysql> select cast("00-00-00 00:00:00.123" as datetime);             
+-------------------------------------------+
| cast("00-00-00 00:00:00.123" as datetime) |
+-------------------------------------------+
| 2000-00-00 00:00:00                       |
+-------------------------------------------+
1 row in set (0.00 sec)
mysql> select cast("1701020304.1" as datetime);                     
+----------------------------------+
| cast("1701020304.1" as datetime) |
+----------------------------------+
| 2017-01-02 03:04:01              |
+----------------------------------+
1 row in set (0.00 sec)
mysql> select cast("17-01-02 03:02:11.23" as datetime); 
+------------------------------------------+
| cast("17-01-02 03:02:11.23" as datetime) |
+------------------------------------------+
| 2017-01-02 03:02:11                      |
+------------------------------------------+
1 row in set (0.00 sec)

I really wonder why mysql cast  "17-01-02 03:02:11.23" to "2017-01-02 03:02:11", dropping fsp part of "23".That is different  from some test cases in time_test.go before I do some change on the file.